File: backend/src/main/java/com/decmoe47/todo/controller/AuthController.java
package com.decmoe47.todo.controller;

import com.decmoe47.todo.model.dto.RefreshTokenDTO;
import com.decmoe47.todo.model.dto.SendVerifyCodeDTO;
import com.decmoe47.todo.model.dto.UserLoginDTO;
import com.decmoe47.todo.model.dto.UserRegisterDTO;
import com.decmoe47.todo.model.vo.AuthenticationTokensVO;
import com.decmoe47.todo.model.vo.R;
import com.decmoe47.todo.model.vo.UserVO;
import com.decmoe47.todo.service.AuthService;
import io.swagger.v3.oas.annotations.Operation;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpHeaders;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final AuthService authService;

    @Operation(summary = "登录账号")
    @PostMapping("/login")
    public R<UserVO> login(@RequestBody UserLoginDTO userLoginDTO, HttpServletResponse response) {
        UserVO userVO = authService.login(userLoginDTO);
        return R.ok(userVO);
    }

    @Operation(summary = "注销")
    @PostMapping("/logout")
    public R<Object> logout(HttpServletRequest request) {
        String token = request.getHeader(HttpHeaders.AUTHORIZATION);
        authService.logout(token);
        return R.ok();
    }

    @Operation(summary = "注册账号")
    @PostMapping("/register")
    public R<UserVO> register(@RequestBody UserRegisterDTO userRegisterDTO) {
        UserVO userVO = authService.register(userRegisterDTO);
        return R.ok(userVO);
    }

    @Operation(summary = "发送验证码")
    @PostMapping("/send-verify-code")
    public R<Object> sendVerifyCode(@RequestBody SendVerifyCodeDTO sendVerifyCodeDTO) {
        authService.sendVerifyCode(sendVerifyCodeDTO.getEmail());
        return R.ok();
    }

    @Operation(summary = "刷新token")
    @PostMapping("/refresh-token")
    public R<AuthenticationTokensVO> refreshToken(@RequestBody RefreshTokenDTO refreshTokenDTO) {
        return R.ok(authService.refreshAccessToken(refreshTokenDTO.getRefreshToken()));
    }
}

File: backend/src/main/kotlin/com/decmoe47/todo/controller/AuthController.kt
package com.decmoe47.todo.controller

import com.decmoe47.todo.model.request.RefreshTokenRequest
import com.decmoe47.todo.model.request.SendVerifyCodeRequest
import com.decmoe47.todo.model.request.UserLoginRequest
import com.decmoe47.todo.model.request.UserRegisterRequest
import com.decmoe47.todo.model.response.AuthenticationTokensResponse
import com.decmoe47.todo.model.response.Response
import com.decmoe47.todo.model.response.UserResponse
import com.decmoe47.todo.service.AuthService
import com.decmoe47.todo.util.R
import io.swagger.v3.oas.annotations.Operation
import jakarta.servlet.http.HttpServletRequest
import jakarta.validation.Valid
import org.springframework.http.HttpHeaders
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/auth")
class AuthController(private val authService: AuthService) {
    @Operation(summary = "登录账号")
    @PostMapping("/login")
    fun login(@Valid @RequestBody request: UserLoginRequest): ResponseEntity<Response<UserResponse>> =
        R.ok(authService.login(request))

    @Operation(summary = "注销")
    @PostMapping("/logout")
    fun logout(servletRequest: HttpServletRequest): ResponseEntity<Response<Unit>> {
        val token = servletRequest.getHeader(HttpHeaders.AUTHORIZATION)
        authService.logout(token)
        return R.ok()
    }

    @Operation(summary = "注册账号")
    @PostMapping("/register")
    fun register(@Valid @RequestBody request: UserRegisterRequest): ResponseEntity<Response<UserResponse>> =
        R.ok(authService.register(request))

    @Operation(summary = "发送验证码")
    @PostMapping("/send-verify-code")
    fun sendVerifyCode(@Valid @RequestBody request: SendVerifyCodeRequest): ResponseEntity<Response<Unit>> {
        authService.sendVerificationCode(request.email)
        return R.ok()
    }

    @Operation(summary = "刷新token")
    @PostMapping("/refresh-token")
    fun refreshToken(@Valid @RequestBody request: RefreshTokenRequest): ResponseEntity<Response<AuthenticationTokensResponse>> =
        R.ok(authService.refreshAccessToken(request.refreshToken))
}
